### PR TITLE
add slice extend methods

### PR DESCRIPTION
**Pull request by UnderTreeTech**

### CODE REVIEW

File: pkg/utils/xslice/float.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+	return false
+}
+
+// RemoveFloat64 remove empty target elements from ss
+func RemoveFloat64(s []float64, target float64) []float64 {
+	var ret = make([]float64, 0)
+	for _, val := range s {
+		if val != target {
+			ret = append(ret, val)
```
Comment (Billxunyang) [inline pkg/utils/xslice/float.go:40]: It seems to append the elements in the original slice to the new slice one by one, which will continue to trigger expansion and affect performance; idea: set the capacity when initializing, insert directly before finding the target, and after finding the target, the left index equal the right index reduce one

File: pkg/utils/xslice/float.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	return ret
+}
+
+// UniqueFloat64 Removes duplicate values from slice
+func UniqueFloat64(s1 []float64) []float64 {
+	unique := make(map[float64]interface{})
+	for _, val := range s1 {
+		unique[val] = nil
+	}
+
```
Comment (Billxunyang) [inline pkg/utils/xslice/float.go:94]: No need to loop twice, once is enough

**Review by Billxunyang — COMMENTED** [review]
2 points can be optimization: function UniqueFloat64 just need to loop once, function  RemoveFloat64 can perform better

File: pkg/utils/xslice/float.go (hunk 2 above)
Comment (UnderTreeTech) [reply pkg/utils/xslice/float.go:94]: If input slice has large size, the overhead of underlying slice expand volume maybe excced saved overhead of one loop.

**Pull request merged**